cli/tools/task.rs

```diff
@@ -112,7 +104,55 @@ pub async fn execute_script(
       false
     }
 
+    // Filter based on package name
+    let package_regex = arg_to_regex(filter)?;
     let workspace = cli_options.workspace();
+
+    let Some(task_name) = &task_flags.task else {
+      let mut matched = false;
+      for folder in workspace.config_folders() {
+        if !matches_package(folder.1, force_use_pkg_json, &package_regex) {
+          continue;
+        }
+        matched = true;
+
+        let member_dir = workspace.resolve_member_dir(folder.0);
+        let mut tasks_config = member_dir.to_tasks_config()?;
+
+        let mut pkg_name = folder
+          .1
+          .deno_json
+          .as_ref()
+          .and_then(|deno| deno.json.name.clone())
+          .or(folder.1.pkg_json.as_ref().and_then(|pkg| pkg.name.clone()));
+
+        if force_use_pkg_json {
+          tasks_config = tasks_config.with_only_pkg_json();
+          pkg_name =
+            folder.1.pkg_json.as_ref().and_then(|pkg| pkg.name.clone());
+        }
+
+        print_available_tasks(
+          &mut std::io::stdout(),
+          &cli_options.start_dir,
+          &tasks_config,
+          pkg_name,
+        )?;
+      }
+
+      if !matched {
+        log::warn!(
+          "{}",
+          colors::red(format!("No package name matched the filter '{}' in available 'deno.json' or 'package.json' files.", filter))
+        );
+      }
+
+      return Ok(0);
```

Comment:
Maybe let's extract out this block to a separate function? It would help to explain what it does.